migrations/v6.js

```diff
@@ -0,0 +1,249 @@
+    content: [
+      { _type: 'course', _boxMenu: {} }
+    ]
+  });
+
+  testStopWhere('incorrect version', {
+    fromPlugins: [{ name: 'adapt-contrib-boxMenu', version: '6.1.0' }]
+  });
+});
+
+describe('Box menu - v6.2.0 to v6.2.1', async () => {
+
+  // https://github.com/adaptlearning/adapt-contrib-boxMenu/compare/v6.2.0..v6.2.1
+
+  let menus;
+  const defaultGraphic = {
+    _src: '',
+    alt: ''
+  };
+
+  whereFromPlugin('Box menu - from v6.2.0', { name: 'adapt-contrib-boxMenu', version: '<6.2.1' });
+
+  whereContent('Box menu - where menus are configured', async (content) => {
+    const candidates = [getCourse(), ...content.filter(({ _type, _component }) => _type === 'menu' && (!_component || _component === 'boxMenu'))];
+    menus = candidates.filter(({ _boxMenu }) => _boxMenu);
+    return menus.length;
+  });
+
+  mutateContent('Box menu - add _graphic attribute', async (content) => {
+    menus.forEach(({ _boxMenu }) => (_boxMenu._graphic = defaultGraphic));
+    return true;
+  });
+
+  checkContent('Box menu - check _graphic attribute', async (content) => {
+    const isValid = menus.every(({ _boxMenu }) => _.isEqual(_boxMenu._graphic, defaultGraphic));
+    if (!isValid) throw new Error('Box menu - course attribute _graphic');
+    return true;
+  });
+
+  updatePlugin('Box menu - update to v6.2.1', { name: 'adapt-contrib-boxMenu', version: '6.2.1', framework: '">=5.24.2' });
+
+  testSuccessWhere('boxMenu with course _boxMenu', {
+    fromPlugins: [{ name: 'adapt-contrib-boxMenu', version: '6.2.0' }],
+    content: [
+      { _type: 'course', _boxMenu: {} },
+      { _type: 'menu', _boxMenu: {} }
+    ]
+  });
+
+  testStopWhere('boxMenu with empty course', {
+    fromPlugins: [{ name: 'adapt-contrib-boxMenu', version: '6.2.0' }],
+    content: [
+      { _type: 'course' },
+      { _type: 'menu' }
+    ]
+  });
+
+  testStopWhere('incorrect version', {
+    fromPlugins: [{ name: 'adapt-contrib-boxMenu', version: '6.2.1' }]
+  });
+});
+
+describe('Box menu - v6.3.8 to v6.3.9', async () => {
+
+  // https://github.com/adaptlearning/adapt-contrib-boxMenu/compare/v6.3.8..v6.3.9
+
+  let menusWithBgImage;
+
+  whereFromPlugin('Box menu - from v6.3.8', { name: 'adapt-contrib-boxMenu', version: '<6.3.9' });
+
+  whereContent('Box menu - where menus have _backgroundImage', async (content) => {
+    const candidates = [getCourse(), ...content.filter(({ _type, _component }) => _type === 'menu' && (!_component || _component === 'boxMenu'))];
+    menusWithBgImage = candidates.filter(({ _boxMenu }) => (
+      _boxMenu?._backgroundImage ||
+      _boxMenu?._menuHeader?._backgroundImage
+    ));
+    return menusWithBgImage.length;
+  });
+
+  mutateContent('Box menu - add _xlarge attribute', async (content) => {
+    menusWithBgImage.forEach(({ _boxMenu }) => {
+      if (_.has(_boxMenu, '_backgroundImage')) {
+        _boxMenu._backgroundImage._xlarge = '';
+      }
```

Comment:
Nesting. Return earlies throughout.

```suggestion
      if (!_.has(_boxMenu, '_backgroundImage')) return;
      _boxMenu._backgroundImage._xlarge = '';
```